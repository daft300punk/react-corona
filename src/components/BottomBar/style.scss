.bottombar {
  position: absolute;
  bottom: 0;
  height: 80px;
  width: 100%;
  background-color: rgba(255, 255, 255, .01);
  border-top: 1px solid rgba(255, 255, 255, .1);
  display: flex;
  flex-direction: row;
  align-items: center;

  div {
    height: 100%;
    flex-grow: 1;
    color: white;
    font-family: sans-serif;
    border-right: 1px solid rgba(255, 255, 255, .1);
    display: flex;
    align-items: center;
    text-align: center;
    justify-content: center;
    font-size: 14px;
    letter-spacing: 1px;

    img {
      height: 40px;
      display: block;
    }
  }

  .category-text {
    cursor: pointer;
  }

  .des::after {
    content: '';
    display: inline-block;
    height: 8px;
    width: 8px;
    border-radius: 50%;
    margin-left: 4px;
    background-color: rgb(40, 169, 224);
    position: relative;
    top: -1px;
  }

  .dsc::after {
    content: '';
    display: inline-block;
    height: 8px;
    width: 8px;
    border-radius: 50%;
    margin-left: 4px;
    background-color: rgb(140, 197, 67);
    position: relative;
    top: -1px;
  }

  .dev::after {
    content: '';
    display: inline-block;
    height: 8px;
    width: 8px;
    border-radius: 50%;
    margin-left: 4px;
    background-color: rgb(246, 147, 34);
    position: relative;
    top: -1px;
  }

  .active {
    border-bottom: 4px solid #28a9e0;
  }

}

@media only screen
  and (max-width: 800px) {
    .bottombar {
      flex-flow: wrap;
      border-top: none;
      background: black;
      position: normal;
      
      .logo {
        position: absolute;
        top: calc(-100vh + 80px);
        border: none;
        width: 100%;
        border-bottom: 2px solid rgba(255, 255, 255, .1);

        img {
          display: inline;
        }
      }

      .logo::after {
        content: 'TOPCODER';
        font-weight: bold;
      }

      .category {
        flex-grow: 0;
        border: none;
        padding: 0 20px 0 20px;
        margin: 0 8px 8px 8px;
        height: 36px;
        border: 1px solid white;
        border-radius: 20px;
      }

      .active {
        background: white;
        color: black;
      }
    }
  }
