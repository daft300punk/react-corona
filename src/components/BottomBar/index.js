import React from 'react';
import './style.scss';
import { categoryTypes } from '../../constants/categoriesAndFilters';
import PropTypes from 'prop-types';

/*
  Bottombar component that appears fixed at the bottom of the screen.
  It holds different available categories. Click on a different 
  category loads the Globe with data for that category.
*/
const BottomBar = ({
  selectedCategory,
  changeSelectedCategory
}) => {

  /*
  A helper function to assist with modifying className on the basis of 
  category text. This is used to render small colored circle marker
  ahead each of Design, Devleopment and Data Science category. The colors
  map to the colors shown on the Globe. So for ex, if data points for Design
  are green in color on the globe, a green circle appears in front of the
  Design category.
  */
  const modifyClassNameHelper = (categoryText) => {
    let className = '';
    if (categoryText == 'design')
      className += 'des';
    if (categoryText == 'data science')
      className += 'dsc';
    if (categoryText == 'development')
      className += 'dev';
    return className;
  };

  return (
    <div className="bottombar">
      <div className="logo">
        <a href="http://www.topcoder.com" target="_blank">
          <img src={require('./logo.png')} />
        </a>
      </div>
      {
        categoryTypes.map((categoryText, i) => {
          let className = 'category ';
          if (selectedCategory === categoryText)
            className += 'active';
          const classSpan = modifyClassNameHelper(categoryText);
          return (
            <div
              key={i}
              onClick={() => { changeSelectedCategory(categoryTypes[i]); }}
              className={'category-text ' + className}
            >
              <span className={classSpan}>{categoryText.toUpperCase()}</span>
            </div>
          );
        })
      }
    </div>
  );
};

BottomBar.propTypes = {
  selectedCategory: PropTypes.string,
  changeSelectedCategory: PropTypes.func
};

export default BottomBar;
