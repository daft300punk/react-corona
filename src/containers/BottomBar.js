import React from 'react';
import { connect } from 'react-redux';
import BottomBar from '../components/BottomBar';
import { changeSelectedCategory } from '../actions';
import PropTypes from 'prop-types';

/*
  See BottomBar component to get the idea. @selectedCategory
  is the currently selected category out of 5 available.
  @changeSelectedCategory is a bound dispathcer that gets
  called when the user clicks on a category other than current.
*/
const BottomBarContainer = ({
  selectedCategory,
  changeSelectedCategory
}) => {
  return (
    <BottomBar
      selectedCategory={selectedCategory}
      changeSelectedCategory={changeSelectedCategory}
    />
  );
};

const mapStateToProps = (state) => ({
  selectedCategory: state.selectedOptions.category
});

const mapDispatchToProps = (dispatch) => ({
  changeSelectedCategory: (filter) => { dispatch(changeSelectedCategory(filter)); }
});

BottomBarContainer.propTypes = {
  selectedCategory: PropTypes.string,
  changeSelectedCategory: PropTypes.func
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BottomBarContainer);
