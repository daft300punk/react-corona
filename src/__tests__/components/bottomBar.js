import React from 'react';
import renderer from 'react-test-renderer';

import BottomBar from '../../components/BottomBar';

describe('Bottombar Component', () => {
  it('renders Bottombar correctly', () => {
    const component = renderer.create(
      <BottomBar
        selectedCategory={'all activity'}
        changeSelectedCategory={() => { }}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});