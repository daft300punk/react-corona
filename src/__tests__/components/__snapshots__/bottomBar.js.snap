exports[`Bottombar Component renders Bottombar correctly 1`] = `
<div
  className="bottombar">
  <div
    className="logo">
    <a
      href="http://www.topcoder.com"
      target="_blank">
      <img
        src="" />
    </a>
  </div>
  <div
    className="category-text category active"
    onClick={[Function]}>
    <span
      className="">
      ALL ACTIVITY
    </span>
  </div>
  <div
    className="category-text category "
    onClick={[Function]}>
    <span
      className="dev">
      DEVELOPMENT
    </span>
  </div>
  <div
    className="category-text category "
    onClick={[Function]}>
    <span
      className="des">
      DESIGN
    </span>
  </div>
  <div
    className="category-text category "
    onClick={[Function]}>
    <span
      className="dsc">
      DATA SCIENCE
    </span>
  </div>
  <div
    className="category-text category "
    onClick={[Function]}>
    <span
      className="">
      TOP WINNERS
    </span>
  </div>
</div>
`;
